### DISCUSSION TITLE
DeviceOrientation API Not Allowed

### DISCUSSION THREAD

**Discussion by Huscarl160** [AR-js-org/AR.js#445] [Help]
Hello. I'm delving into AR.js for a project and I need a reliable location-based AR solution and I believe AR.js is the near-perfect solution. My problem is two-fold. 

1) When I use the A-Frame version, it works, but the model seems to be locked to my phone (iPhone 12, iOS: 15.5, Chrome). I want to see the model at the Lat/Long coordinates, but it sits at my phone. I've read this is a possible issue with iOS as they have not been tested as much. One suggestion was to use the 3JS version. Which brings me to...

2) Using the 3JS version, I can not get my model to display. Running a Console log, I get "ERROR: THREE.DeviceOrientationControls: Unable to use DeviceOrientation API" It seems to me that it's not able to invoke iOS to grant it permission to use its sensors to establish its orientation. It asks me for permission to use the camera, but not the sensors.

Does anyone have a suggestion? Thank you!
![ERROR DeviceOrientationControl](https://user-images.githubusercontent.com/82227975/181584676-bd91aaf5-331f-459d-b887-84db72d27aa8.png)

**Comment by nickw1** [comment]
Apologies for the delay in answering. These are iOS specific issues. We have had some PRs recently related to iOS; if you update to the latest version (3.4.2) does it work?
